contributed/rho_statistics.ipynb

```diff
@@ -0,0 +1,1091 @@
+    "                              (2.*lt[1,0]*lt[1,1])**2*tcyy + lt[1,1]**4*tcxy)\n",
+    "                cov_ixy[i] = ((lt[0,0]*lt[1,0])**2*tcxx +\n",
+    "                              (lt[0,0]*lt[1,1]+lt[0,1]*lt[1,0])**2*tcyy +\n",
+    "                              (lt[0,1]*lt[1,1])**2*tcxy)\n",
+    "            return cov_ixx, cov_iyy, cov_ixy\n",
+    "        return Cxx, Cyy, Cxy\n",
+    "    \n",
+    "    def g1(self):\n",
+    "        return (self.Ixx-self.Iyy)/(self.Ixx+self.Iyy)\n",
+    "\n",
+    "    def g2(self):\n",
+    "        return 2.*self.Ixy/(self.Ixx+self.Iyy)\n",
+    "    \n",
+    "    def sigma(self):\n",
+    "        return 3600*(self.Ixx*self.Iyy - self.Ixy**2)**0.25 #3600: in arcsec\n",
+    "    \n",
+    "    def g1_err(self):\n",
+    "        dg1_dixx = 2.*self.Iyy/(self.Ixx+self.Iyy)**2\n",
+    "        dg1_diyy = -2.*self.Ixx/(self.Ixx+self.Iyy)**2\n",
+    "        return np.sqrt(dg1_dixx**2 * self.Cxx + dg1_diyy**2 * self.Cyy)\n",
+    "\n",
+    "    def g2_err(self):\n",
+    "        dg2_dixx = -2.*self.Ixy/(self.Ixx+self.Iyy)**2\n",
+    "        dg2_diyy = -2.*self.Ixy/(self.Ixx+self.Iyy)**2\n",
+    "        dg2_dixy = 2./(self.Ixx+self.Iyy)\n",
+    "        return np.sqrt(dg2_dixx**2 * self.Cxx + dg2_diyy**2 * self.Cyy +\n",
+    "                            dg2_dixy**2 * self.Cxy)\n",
+    "\n",
+    "    def sigma_err(self):\n",
+    "        sigma = self.sigma()\n",
+    "        dsigma_dixx = 0.25/sigma**3*self.Iyy\n",
+    "        dsigma_diyy = 0.25/sigma**3*self.Ixx\n",
+    "        dsigma_dixy = -0.5/sigma**3*self.Ixy\n",
+    "        return 3600*np.sqrt(dsigma_dixx**2 * self.Cxx + dsigma_diyy**2 * self.Cyy +\n",
+    "                               dsigma_dixy**2 * self.Cxy)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Grabbing catalogs\n",
+    "-----------------\n",
+    "\n",
+    "Next, let's get catalogs to do some analysis. We'll want four different catalogs:\n",
+    "1. The extragalactic catalog\n",
+    "2. The truth catalog (for matching positions)\n",
+    "3. An object catalog (from DM processing) with galaxy shapes and PSF shapes at the locations of the galaxies\n",
+    "4. An object catalog with star shapes and PSF shapes at the locations of the stars\n",
+    "\n",
+    "We'll select these catalogs on a tract-by-tract basis.  By default, we will use tract 4851; for the run 1.2p that we are using, we could use tracts 4429-4433, 4636-4640, 4848-4852, or 5062-5066 (or a list of any or all of those--note that the runtime of this notebook goes up with the number of tracts, though).  To limit the catalogs to the correct area, we use code from [this tutorial by Jim Chiang](https://nbviewer.jupyter.org/github/LSSTDESC/DC2-analysis/blob/rendered/tutorials/matching_stack.nbconvert.ipynb).  To do the shear rotation defined above, and the tract selection here, we will use a DM stack butler, as mentioned above."
+   ]
```

Comment:
How easy is this to port to run 2?

Reply:
My memory is that it was going to be very annoying to do--some differently named columns, some different access functions for things. And since I'm starting an industry job in January (er, surprise?) I don't think I have time to do this, at this point, although I'm happy for someone else to take it on.